adapter/test/mpiio/mpiio_adapter_test.cpp

```diff
@@ -0,0 +1,671 @@
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include <adapter_utils.h>
+#include <catch_config.h>
+#include <unistd.h>
+
+#include <experimental/filesystem>
+#include <iostream>
+
+#if HERMES_INTERCEPT == 1
+#include <hermes/adapter/mpiio.h>
+#endif
+
+namespace fs = std::experimental::filesystem;
+
+namespace hermes::adapter::mpiio::test {
+struct Arguments {
+  std::string filename = "test.dat";
+  std::string directory = "/tmp";
+  size_t request_size = 65536;
+};
+struct Info {
+  int rank = 0;
+  int comm_size = 1;
+  std::string write_data;
+  std::string read_data;
+  std::string new_file;
+  std::string existing_file;
+  std::string shared_new_file;
+  std::string shared_existing_file;
+  std::string new_file_cmp;
+  std::string existing_file_cmp;
+  std::string shared_new_file_cmp;
+  std::string shared_existing_file_cmp;
+  size_t num_iterations = 64;
+  unsigned int offset_seed = 1;
+  unsigned int rs_seed = 1;
+  unsigned int temporal_interval_seed = 5;
+  size_t total_size;
+  size_t stride_size = 512;
+  unsigned int temporal_interval_ms = 1;
+  size_t small_min = 1, small_max = 4 * 1024;
+  size_t medium_min = 4 * 1024 + 1, medium_max = 256 * 1024;
+  size_t large_min = 256 * 1024 + 1, large_max = 3 * 1024 * 1024;
+};
+}  // namespace hermes::adapter::mpiio::test
+hermes::adapter::mpiio::test::Arguments args;
+hermes::adapter::mpiio::test::Info info;
+std::string gen_random(const int len) {
+  std::string tmp_s;
+  static const char alphanum[] =
+      "0123456789"
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+      "abcdefghijklmnopqrstuvwxyz";
+
+  srand(100);
+
+  tmp_s.reserve(len);
+
+  for (int i = 0; i < len; ++i)
+    tmp_s += alphanum[rand() % (sizeof(alphanum) - 1)];
+
+  return tmp_s;
+}
+int init(int* argc, char*** argv) {
+  MPI_Init(argc, argv);
+  info.write_data = gen_random(args.request_size);
+  info.read_data = std::string(args.request_size, 'r');
+  MPI_Comm_rank(MPI_COMM_WORLD, &info.rank);
+  MPI_Comm_size(MPI_COMM_WORLD, &info.comm_size);
+  return 0;
+}
+int finalize() {
+  MPI_Finalize();
+  return 0;
+}
+
+int pretest() {
+  fs::path fullpath = args.directory;
+  fullpath /= args.filename;
+  info.new_file = fullpath.string() + "_new_" + std::to_string(getpid());
+  info.existing_file = fullpath.string() + "_ext_" + std::to_string(getpid());
+  info.new_file_cmp =
+      fullpath.string() + "_new_cmp" + "_" + std::to_string(getpid());
+  info.existing_file_cmp =
+      fullpath.string() + "_ext_cmp" + "_" + std::to_string(getpid());
+  info.shared_new_file =
+      fullpath.string() + "_new_" + std::to_string(info.comm_size);
+  info.shared_existing_file =
+      fullpath.string() + "_ext_" + std::to_string(info.comm_size);
+  info.shared_new_file_cmp =
+      fullpath.string() + "_new_cmp_" + std::to_string(info.comm_size);
+  info.shared_existing_file_cmp =
+      fullpath.string() + "_ext_cmp_" + std::to_string(info.comm_size);
```

Comment:
Do these filenames need a pid appended?

Reply:
I see that they need a uniform name so that they can be shared by each process, but we still run into the issue where one person creates these files in `/tmp` on Ares and then the next person to run the tests gets permission errors.

Reply:
Appended the USER variable to the filename.